sdk/storage/AzureStorageBlob/Source/Operation/TransferOperationQueue.swift

```diff
@@ -34,27 +34,15 @@ internal class TransferOperationQueue: OperationQueue {
     }
 
     func add(_ operations: [TransferOperation]) {
-        let states = operations.map { $0.transfer.state.rawValue }
-        for state in Set(states) {
-            let filteredOps = operations.filter { $0.transfer.state.rawValue == state }
-            var transferState = TransferState(rawValue: state)!
-            let allowed: [TransferState] = [.pending, .inProgress]
-            if allowed.contains(transferState) {
-                // reset inProgress back to pending since they may be scheduled differently
-                transferState = .pending
-
-                // For Pending or InProgress operations, need to requeue the operations
-                for operation in filteredOps {
-                    let operationClosure = operation.completionBlock
-                    operation.completionBlock = {
-                        operationClosure?()
-                        if !operation.transfer.isActive {
-                            return
-                        }
-                    }
-                }
-                addOperations(filteredOps, waitUntilFinished: false)
+        let filteredOps = operations.filter { $0.transfer.state.active }
```

Comment:
Is this the correct filter? What states must a transfer be in, for them to be eligible to be added to the operation queue? My guess would be pending, failed, and paused.

Reply:
According to the prior logic, only `.pending` and `.inProgress` transfers would be permitted. This simplifies that to use `state.active`. For failed and paused operations, they would be put into pending state prior to this method being called.